src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1518,7 +1518,10 @@ private File createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOExcepti
         PrintWriter w = null;
         try {
             w = new PrintWriter(ssh);
-            w.println("echo \"" + Secret.toString(sshUser.getPassphrase()) + "\"");
+            // avoid echoing command as part of the password
+            w.println("@echo off");
+            // no need for quotes on windows echo -- they will get echoed too
+            w.println("echo " + Secret.toString(sshUser.getPassphrase()));
```

Comment:
Eek. Definitely don't want to open up a code injection vulnerability. :cry:

And yes, the key we are using does indeed have a passphrase so that the private key cannot be grabbed from the jenkins UI. That way the private is not usable without gaining access to the jenkins server and grabbing the (obfuscated) passphrase. 

I wonder if it would be better to be changed to something like: ```cmd /C "echo foo"``` together with escaping quotes (or perhaps not allowing quotes?) in the passphrase itself.